ciere/json/detail/value_impl.hpp

```diff
@@ -45,6 +45,7 @@ namespace ciere { namespace json
          static R apply( T const & v
                        , typename boost::enable_if<boost::is_convertible<R,T> >::type* dummy=0 )
          {
+            (void)dummy;
             return v;
          }
 
```

Comment:
A more typical approach we use is to simply comment out the argument. E.g. /*dummy*/=0. That would be less cluttered than the solution here.